database/src/main/scala/cromwell/database/migration/metadata/MetadataMigration.scala

```diff
@@ -0,0 +1,76 @@
+package cromwell.database.migration.metadata
+
+import java.sql.{PreparedStatement, ResultSet}
+
+import cromwell.database.migration.ResultSetIterator
+import liquibase.change.custom.CustomTaskChange
+import liquibase.database.Database
+import liquibase.database.jvm.JdbcConnection
+import liquibase.exception.{CustomChangeException, ValidationErrors}
+import liquibase.resource.ResourceAccessor
+import org.slf4j.LoggerFactory
+
+import scala.language.postfixOps
+
+trait MetadataMigration extends CustomTaskChange {
+  val logger = LoggerFactory.getLogger("LiquibaseMetadataMigration")
+
+  protected def selectQuery: String
+  protected def migrateRow(connection: JdbcConnection, collectors: Set[Int],
+                           statement: PreparedStatement, row: ResultSet, idx: Int): Unit
+
+  private def migrate(connection: JdbcConnection, collectors: Set[Int]) = {
+    val executionDataResultSet = connection.createStatement().executeQuery(selectQuery)
+    val metadataInsertStatement = MetadataStatement.makeStatement(connection)
+
+    val executionIterator = new ResultSetIterator(executionDataResultSet)
+
+    executionIterator.zipWithIndex foreach {
+      case (row, idx) =>
+        migrateRow(connection, collectors, metadataInsertStatement, row, idx)
+        if (idx % 100 == 0) {
+          metadataInsertStatement.executeBatch()
+          connection.commit()
+        }
+    }
+
+    metadataInsertStatement.executeBatch()
+    connection.commit()
```

Comment:
This was a problem on Hsql until 2.3.4 because they wouldn't execute empty batches, but they added a flag to allow so it should be fine now.